src/gpu/nvidia/sycl_cuda_memory_storage_helper.hpp

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+* Copyright 2022 Codeplay Software Limited
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_NVIDIA_SYCL_CUDA_MEMORY_STORAGE_HELPER_HPP
+#define GPU_NVIDIA_SYCL_CUDA_MEMORY_STORAGE_HELPER_HPP
+
+#include <optional>
+#include "gpu/nvidia/sycl_cuda_scoped_context.hpp"
+#include "sycl/sycl_memory_storage.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace nvidia {
+
+#define CTX_IN_MEMORY(arg) \
+    static_cast<sycl::sycl_memory_storage_base_t *>(&CTX_IN_STORAGE(arg))
+
+#define CTX_OUT_MEMORY(arg) \
+    static_cast<sycl::sycl_memory_storage_base_t *>(&CTX_OUT_STORAGE(arg))
+
+#define CTX_IN_OPTIONAL_ACCESSOR(arg, memory) \
+    get_cudnn_accessor<decltype(CTX_IN_ACCESSOR(arg))>(memory, cgh)
```

Comment:
How do you look at moving `CTX_IN_ACCESSOR` and related macros to this file since they refer semantically to the same thing?

Reply:
I would suggest delaying this change until USM support is added to all cudnn operations in oneDNN. The reason being is that at the moment almost all operations are using this. So such a refactor would require swapping the include of `sycl_cuda_utils.hpp` to `sycl_cuda_memory_storage_helper.hpp` in ~35 files. That is something I would prefer putting in a separate pull request anyway. 

Most of these operations will not need to directly include `sycl_cuda_utils.hpp` when they support USM anyway, so the refactor would be simpler after.

Reply:
Gotcha, resolving.